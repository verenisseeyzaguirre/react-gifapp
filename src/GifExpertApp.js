// rafc
import React, { useState } from 'react'

import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = () => {

    //const categories = ['Panda', 'Sailon Moon', 'gatos']
    const [categories, setCategories] = useState(['Panda']);

    // const handleAdd = () => {
    //     /*setCategories([...categories , 'Candy'])*/
    //    setCategories( cats => [...cats , 'Mon'] )
    // }

    return (
        <>
            <h2>Gif App</h2>
            <p>Encuentra los mejores gifs</p>
            <AddCategory setCategories= { setCategories } />
            <hr/>

            <ol>
                {
                    categories.map( category => (
                        <GifGrid
                            key = {category}
                            category = {category}
                        />
                    ))
                }
            </ol>
        </>
    )
}

/*
    categories.map( category => {
        return <li key={ category } > { category } </li>
    })
*/