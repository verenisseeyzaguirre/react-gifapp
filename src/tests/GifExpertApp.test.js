import React from 'react'
import { shallow } from 'enzyme';

import { GifExpertApp } from '../GifExpertApp';


describe('Pruebas en <GifGrid />', ( ) => {

    test( 'Debe mostrar <GifExpertApp /> correctamente' , () => {
    
        const wrapper = shallow( <GifExpertApp /> )
        expect( wrapper ).toMatchSnapshot();

    });

    test( 'Debe mostrar una lista de categorias' , () => {
    
        const categories = ['Panda' , 'Sailor Moon'];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } /> )
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length )

    });

})