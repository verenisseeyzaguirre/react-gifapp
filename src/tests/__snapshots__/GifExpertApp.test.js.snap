// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe mostrar <GifExpertApp /> correctamente 1`] = `
<Fragment>
  <h2>
    Gif App
  </h2>
  <p>
    Encuentra los mejores gifs
  </p>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    Gif App
  </h2>
  <p>
    Encuentra los mejores gifs
  </p>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Panda"
      key="Panda"
    />
    <GifGrid
      category="Sailor Moon"
      key="Sailor Moon"
    />
  </ol>
</Fragment>
`;
